build/gulp/tasks/docs.ts

```diff
@@ -90,7 +90,11 @@ task('build:docs:images', () =>
 task('build:docs:toc', () =>
   src(markdownSrc, { since: lastRun('build:docs:toc') }).pipe(
     through2.obj((file, enc, done) => {
-      sh(`doctoc ${file.path} --github --maxlevel 4 && git add ${file.path}`, done)
+      sh(`doctoc ${file.path} --github --maxlevel 4`, err => {
+        if (err) return done(err)
+
+        sh(`git add ${file.path}`, done)
```

Comment:
pretty sure that this was a deliberate decision to split these two shell sessions on a separate processes. Could you, please, suggest what was the reason for that? (just for the sake of curiosity, see no problems with that)

Reply:
`sh` was refactored to use `spawn` instead of `exec`.  This refactor allowed us to pipe the output of spawned tasks back to the main stream so we can see what tasks are doing.

The signature for `spawn` accepts a command and arguments, we can no longer call sh with `&&` to concat commands as this was previously doing:

```tsx
sh('echo "hi" && echo "done") // worked with child_process.exec()
sh('echo "hi" && echo "done") //  fails with child_process.spawn()
```

Instead of trying to magically split the commands on `&&`, I opted to only allow one command per call to `sh` for now.